src/Web/Sfa.Das.Sas.Shared.Components/Mapping/TrainingProviderClosestLocationsViewModelMapper.cs

```diff
@@ -0,0 +1,35 @@
+﻿using Sfa.Das.Sas.ApplicationServices.Responses;
+using Sfa.Das.Sas.Core.Domain.Model;
+using Sfa.Das.Sas.Shared.Components.ViewModels;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Sfa.Das.Sas.Shared.Components.Mapping
+{
+    public class TrainingProviderClosestLocationsViewModelMapper : ITrainingProviderClosestLocationsViewModelMapper
+    {
+        public ClosestLocationsViewModel Map(string apprenticeshipId, int ukprn, string postcode, GetClosestLocationsResponse source)
+        {
+            var model = new ClosestLocationsViewModel
+            {
+                ApprenticeshipId = apprenticeshipId,
+                ProviderName = source.ProviderName,
+                Ukprn = ukprn,
+                Locations = source.Results.Hits.Select(x => new CloseLocationViewModel { Distance = x.Distance, PostCode = x.Address.Postcode, AddressWithoutPostCode = GetCommaDelimitedAddress(x.Address) }).ToList()
+            };
+
+            return model;
+        }
+
+        private string GetCommaDelimitedAddress(Address address)
+        {
+            var addressParts = new List<string>(5);
```

Comment:
Should this be moved to the address class as used in more than one class?

Reply:
Address is a domain model. I think this logic is very UI specific. Probably better suited in an Extension method defined in UI project.